### PR TITLE
Master to 0.4.0

### PR DESCRIPTION
**Pull request by lirantal**
This is 0.4.0 branch after I merged it with changes from master and fixes.
Once we review it and want to continue forward with move to 0.4.0:

1. We will merge this PR with 0.4.0
2. We will merge any outstanding and important 0.4.0 PRs with 0.4.0
3. We will merge 0.4.0 into master and make it an 0.4.0 release

### CODE REVIEW

**Comment by lirantal** [comment]
@ilanbiala @roieki @amoshaviv @mleanos please review before we merge this into 0.4.0 and further on to master

File: .editorconfig [hunk 1]
```diff
@@ -33,4 +33,4 @@ indent_style = tab
 
 # Standard at: 
 [Makefile]
-indent_style = tab
+indent_style = tab
```
Comment (ilanbiala) [inline .editorconfig]: This should have a newline

File: .gitignore [hunk 2]
```diff
@@ -1,13 +1,66 @@
...
+ehthumbs.db
+Icon?
+Thumbs.db
+config/env/local.js
+
+# Node and related ecosystem
+# ==========================
 .nodemonignore
 .sass-cache/
-*.log
```
Comment (ilanbiala) [inline .gitignore]: I thought we said this is better

File: .travis.yml [hunk 3]
```diff
@@ -1,6 +1,7 @@
 language: node_js
 node_js:
   - "0.10"
+  - "0.11"
```
Comment (ilanbiala) [inline .travis.yml]: We should change this to 0.12 now

File: modules/core/client/views/header.client.view.html [hunk 4]
```diff
@@ -59,4 +59,4 @@
 			</li>
 		</ul>
 	</nav>
-</div>
+</div>
```
Comment (ilanbiala) [inline modules/core/client/views/header.client.view.html]: Newline missing

File: modules/core/server/views/layout.server.view.html [hunk 5]
```diff
@@ -69,17 +70,6 @@
...
-				m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
-		})(window,document,'script','//www.google-analytics.com/analytics.js','ga');
-
-		ga('create', '{{googleAnalyticsTrackingID}}', 'auto');
-		ga('send', 'pageview');
-	</script>
 </body>
 
-</html>
+</html>
```
Comment (ilanbiala) [inline modules/core/server/views/layout.server.view.html]: Newline needed

File: modules/users/server/config/users.server.config.js [hunk 6]
```diff
@@ -31,4 +34,4 @@ module.exports = function(app, db) {
 	// Add passport's middleware
 	app.use(passport.initialize());
 	app.use(passport.session());
-};
+};
```
Comment (ilanbiala) [inline modules/users/server/config/users.server.config.js]: Newline needed

File: package.json [hunk 7]
```diff
@@ -95,4 +95,4 @@
 		"karma-firefox-launcher": "~0.1.3",
 		"karma-phantomjs-launcher": "~0.1.2"
 	}
-}
+}
```
Comment (ilanbiala) [inline package.json]: Newline needed

File: server.js [hunk 8]
```diff
@@ -16,5 +22,13 @@ mongoose.connect(function (db) {
...
+	console.log('--');
+	console.log(chalk.green(config.app.title + ' application started'));
+	console.log(chalk.green('Environment:\t\t\t' + process.env.NODE_ENV));
+	console.log(chalk.green('Port:\t\t\t\t' + config.port));
+	console.log(chalk.green('Database:\t\t\t' + config.db.uri));
+	if (process.env.NODE_ENV === 'secure') {
+		console.log(chalk.green('HTTPs:\t\t\t\ton'));
+	}
+	console.log('--');
+});
```
Comment (ilanbiala) [inline server.js]: Newline needed

**Comment by ilanbiala** [comment]
@lirantal @roieki @amoshaviv @NeverOddOrEven @Shrizzy Left some comments to review.

File: .gitignore (hunk 2 above)
Comment (lirantal) [inline .gitignore]: there's a *.log at the end of the .gitignore file in the 'general' section

File: .travis.yml (hunk 3 above)
Comment (lirantal) [inline .travis.yml]: I'll just add 0.12 as well

**Comment by lirantal** [comment]
thanks @ilanbiala I addressed all issues you mentioned

File: .gitignore (hunk 2 above)
Comment (ilanbiala) [inline .gitignore]: Then we should remove it

File: .travis.yml (hunk 3 above)
Comment (ilanbiala) [inline .travis.yml]: I think 0.11 just redirects to 0.12, so just change it maybe?

File: .gitignore (hunk 2 above)
Comment (lirantal) [inline .gitignore]: @ilanbiala can you be more clear? we have a *.log at the end of the .gitignore and that should cover all such files as npm-debug.log, and I'll now commit to remove access.log too.

File: .travis.yml (hunk 3 above)
Comment (lirantal) [inline .travis.yml]: Cool, so we want to keep 0.10 and 0.12, right?

Comment (ilanbiala) [inline .travis.yml]: Yes

**Comment by lirantal** [comment]
@ilanbiala I think we're good to merge this to 0.4, right?

**Comment by ilanbiala** [comment]
Yes, I think so.

**Comment by mleanos** [comment]
Had a holiday weekend, so I spend much time at my computer over the last few days.

So far it looks good to me. I'm still going through it right now.

**Comment by rhutchison** [comment]
Update dependcies: https://github.com/meanjs/mean/pull/622

**Comment by mleanos** [comment]
All seems fine with me as well. I think we can start merging this into 0.4

**Comment by mleanos** [comment]
I forked this repository. I then checked the master_to_0.4.0 branch out, and tested locally. Everything went fine, and I merged it to the 0.4.0. The merge had no issues at all, and everything seems to be working perfectly.

I'm not sure what I should do next to help this merge, and then to begin helping out with merging all the PR's.

Forgive me, but I'm a bit new to the forking and contributing aspect of GitHub. Should I push my new 0.4.0 branch to my repo, and create a pull request? I did make one modification to the gitingore. I added VS files.

**Comment by lirantal** [comment]
@mleanos thanks for helping out.
The procedure you did confirms that we shouldn't have any issues with the merge, but basically the plan is now:
1. Merge this PR master_to_0.4.0 to 0.4.0
2. Then merge any valid and outstanding 0.4.0 PRs to 0.4.0
3. Then we make a 0.4.0 release and merge 0.4.0 to master

At the moment I'm just waiting to figure out if we're merging this PR https://github.com/meanjs/mean/pull/622 to master_to_0.4.0 before I'm closing it, and then we can continue.

@mleanos once step (2) is finished we'll need quite some help with testing the 0.4.0 branch and that's where you can help the most. It'll be a few days/week until we're there.

Thanks for chiming in!

**Comment by lirantal** [comment]
Finally merging this to 0.4.0

File: scripts/generate-ssl-certs.sh [hunk 9]
```diff
@@ -1,11 +1,17 @@
...
+fi
+
 echo "Generating self-signed certificates..."
+mkdir -p ./config/sslcerts
 openssl genrsa -out ./config/sslcerts/key.pem -aes256 1024
 openssl req -new -key ./config/sslcerts/key.pem -out ./config/sslcerts/csr.pem
 openssl x509 -req -days 9999 -in ./config/sslcerts/csr.pem -signkey ./config/sslcerts/key.pem -out ./config/sslcerts/cert.pem
 rm ./config/sslcerts/csr.pem
-# resolve issue with bad password...
-# Error: error:0906A068:PEM routines:PEM_do_header:bad password read
```
Comment (jloveland) [inline scripts/generate-ssl-certs.sh]: @lirantal for some reason I get this error running with ssl after generating a cert with a password:
`Error: error:0906A068:PEM routines:PEM_do_header:bad password read`

my suspicion is that node is not asking me for the password when running `grunt prod`. I am running with node v0.12.5

to resolve, I needed to remove the password from the pem like so:
```
openssl rsa -in ./config/sslcerts/key.pem -out ./config/sslcerts/newkey.pem && mv ./config/sslcerts/newkey.pem ./config/sslcerts/key.pem
chmod 0600 ./config/sslcerts/key.pem ./config/sslcerts/cert.pem
```
not sure if anyone else has seen this, but I mentioned the a reference to another person solving this issue the same way:
http://blog.mgechev.com/2014/02/19/create-https-tls-ssl-application-with-express-nodejs/

should I open a ticket for this?

Comment (lirantal) [inline scripts/generate-ssl-certs.sh]: @jloveland yes, please open an issue on that and assign it to me, I'll get to it later this week.

**Pull request merged**